### PR TITLE
ZKR-2588-Scroll-Support

### PR DESCRIPTION
**Pull request by FatemehHeidari**
This PR introduces support for Scroll, accommodating the unique handling of lookup arguments in Scroll compared to other versions of halo2. To address this, we've implemented two versions of the `decompose_lookups()` function, each tailored to the distinct structures encountered in Scroll and other halo2 versions.

Key Highlights:

- Scroll Support: Ensures compatibility with Scroll's approach to lookup arguments, enhancing the analyzer's versatility across different halo2 implementations.
- Permutation Optimization Fix: Addresses an issue where the SMT solver constraints included redundant equality constraints with identical values on both sides. This update removes such redundancies.
- Test Improvements: Incorporates general fixes and enhancements to the testing suite.

### CODE REVIEW

File: benchmarks/src/benchmark_bit_decomp.rs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+//     let start = Instant::now();
+//     let _result = analyzer.analyze_underconstrained(analyzer_input, &prime);
+//     let duration = start.elapsed();
+
+//     println!(
+//         "{} bits: Time elapsed for analyze_underconstrained() is: {:?}",
+//         BITS, duration
+//     );
+// }
+//TODO: Add Benchmark tests for bit decomposition later
```
Comment (jgorzny) [inline benchmarks/src/benchmark_bit_decomp.rs:94]: Again, let's have a ticket referenced here and no dead code files in the repo.

**Review by jgorzny — CHANGES_REQUESTED** [review]
This looks like it contains most of the changes of https://github.com/quantstamp/halo2-analyzer/pull/33 too

File: benchmarks/src/benchmark_bit_decomp.rs (hunk 1 above)
Comment (FatemehHeidari) [reply benchmarks/src/benchmark_bit_decomp.rs:94]: Fixes [here](https://github.com/quantstamp/halo2-analyzer/pull/33/commits/ab9c7c09d5e26744f5938419cebd61c23226e301).

Comment (jgorzny) [reply benchmarks/src/benchmark_bit_decomp.rs:94]: Ok; let's merge that one in first

Comment (FatemehHeidari) [reply benchmarks/src/benchmark_bit_decomp.rs]: Done! Now the changes all related to the current PR.

**Review by jgorzny — APPROVED** [review]

**Pull request merged**